### PR TITLE
Update plotting to handle prediction intervals

### PR DESCRIPTION
**Pull request by timcashion**
#### Reference Issues/PRs
Fixes Issue [#541](https://github.com/alan-turing-institute/sktime/issues/541)

#### What does this implement/fix? Explain your changes.
Allows plot_series to take pred_int (2nd output of model.predict()) as an option variable. If present, plots lower and upper bound confidence intervals for prediction using plt.fill_between. The index of pred_int is also checked to make sure it fits within same axis as existing series. 

#### Does your contribution introduce a new dependency? If yes, which one?
No

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/master/.all-contributorsrc).
- [ ] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/alan-turing-institute/sktime/blob/master/CODEOWNERS) to receive notifications about future changes to these files.
- [ ] I've added unit tests and made sure they pass locally.

### CODE REVIEW

File: sktime/utils/plotting/__init__.py [hunk 1]
```diff
@@ -75,6 +77,25 @@ def plot_series(*series, labels=None):
 
         plot_func(x=x, y=y, ax=ax, marker="o", label=label, color=color)
 
+    # plot prediction intervals if present
+    if pred_int is not None:
+        # check same conditions as for earlier indices
+        if not type(index) is type(pred_int.index):  # noqa
+            raise TypeError("Found series with different index types.")
+        elif all([x in index for x in pred_int.index]):
+            plt.fill_between(
```
Comment (mloning) [inline sktime/utils/plotting/__init__.py:86]:
```suggestion
            ax.fill_between(
```

File: sktime/utils/plotting/__init__.py [hunk 2]
```diff
@@ -75,6 +77,25 @@ def plot_series(*series, labels=None):
...
+                pred_int.index,
+                pred_int.lower,
+                pred_int.upper,
+                alpha=0.3,
+                color=ax.get_lines()[-1].get_c(),
+                label="95% prediction intervals",
```
Comment (mloning) [inline sktime/utils/plotting/__init__.py:92]: Alpha may be different so we should assign a more general label here:
```suggestion
                label="prediction intervals"
```

**Review by mloning — COMMENTED** [review]
Hi @timcashion - thanks for contributing! I had a quick look and something still seems to not work as expected. 

Adding a simple unit test that calls the function would be nice too. 

When I run this, it throws an error: 
```python
from sktime.forecasting.all import *
y = load_airline()
y_train, y_test = temporal_train_test_split(y, test_size=36)
fh = ForecastingHorizon(y_test.index, is_relative=False)
forecaster = ThetaForecaster(sp=12)
forecaster.fit(y_train)
y_pred, pred_int = forecaster.predict(fh, return_pred_int=True, alpha=0.05)
fig, ax = plot_series(y_train, y_test, y_pred, 
                      labels=["y_train", "y_test", "y_pred"], pred_int=pred_int)
```
Seems like it doesn't like the `pred_int.index` for the x-values in `fill_between`.

File: sktime/utils/plotting/__init__.py (hunk 2 above)
Comment (timcashion) [reply sktime/utils/plotting/__init__.py:92]: Makes sense!

**Comment by mloning** [comment]
I think we need to use this line something similar to what we do in this line: `xs = [np.argwhere(index.isin(y.index)).ravel() for y in series]` to get the x values for the prediction intervals.

**Comment by timcashion** [comment]
Added two unit tests for plot_series, and this now works with integer and datetime indices.

**Comment by mloning** [comment]
@timcashion have you forgotten to push the latest changes, can't see them?

**Comment by mloning** [comment]
Closed in favour of #802
